mcs/class/System/Mono.Net.Security/MobileTlsContext.cs

```diff
@@ -54,6 +55,13 @@ abstract class MobileTlsContext : IDisposable
 			this.clientCertificates = clientCertificates;
 			this.askForClientCert = askForClientCert;
 
+			serverName = targetHost;
+			if (!string.IsNullOrEmpty (serverName)) {
+				var pos = serverName.IndexOf (':');
```

Comment:
It does not.  

The ServerName extension is designed to share a single IP address with many different host names, so the server name must be a name ;-)

It is in the spec:

https://tools.ietf.org/html/rfc3546#page-8